### PR TITLE
[REF] web: convert command_service tests to hoot

### PR DESCRIPTION
**Pull request by kebeclibre**
task-3705027

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kebeclibre** [comment]
Wait for  https://github.com/odoo/odoo/pull/152421
replace keyUp/keyDown (control+k) with press("control.k")

File: addons/web/static/tests/core/commands/command_service.test.js [hunk 1]
```diff
@@ -0,0 +1,1214 @@
...
+        }
+    });
+    commandSetupRegistry.getEntries().forEach(([key]) => {
+        if (!["command", "data-hotkeys", "default"].includes(key)) {
+            commandSetupRegistry.remove(key);
+        }
+    });
+    env = await makeMockEnv();
+    const commandCategoryRegistry = registry.category("command_categories");
+    // Adding default last. The order of of insertion of categories matters
```
Comment (aab-odoo) [inline addons/web/static/tests/core/commands/command_service.test.js:43]: of of

File: addons/web/static/tests/core/commands/command_service.test.js [hunk 2]
```diff
@@ -0,0 +1,1214 @@
...
+}
+
+const commandProviderRegistry = registry.category("command_provider");
+const commandSetupRegistry = registry.category("command_setup");
+
+let env;
+beforeEach(async () => {
+    commandProviderRegistry.getEntries().forEach(([key]) => {
+        if (!["command", "data-hotkeys", "default"].includes(key)) {
+            commandProviderRegistry.remove(key);
```
Comment (aab-odoo) [inline addons/web/static/tests/core/commands/command_service.test.js:33]: what are you removing here? You don't re-add them after...

File: addons/web/static/tests/core/commands/command_service.test.js [hunk 3]
```diff
@@ -0,0 +1,1214 @@
...
+        setup() {
+            useCommand("Take the throne", () => {
+                expect.step("Hodor");
+            });
+        }
+    }
+    const componentInfo = reactive({ Component: MyComponent });
+    await mountWithCleanup(Parent, { props: { componentInfo } });
+
+    // hoot-event: press doesn't work as expected for this
```
Comment (aab-odoo) [inline addons/web/static/tests/core/commands/command_service.test.js:80]: what do you mean?

File: addons/web/static/tests/core/commands/command_service.test.js [hunk 4]
```diff
@@ -0,0 +1,1214 @@
...
+});
+
+test("command with hotkey", async () => {
+    const hotkey = "a";
+    env.services.command.add("test", () => expect.step(hotkey), {
+        hotkey,
+    });
+    await animationFrame();
+
+    press("a");
```
Comment (aab-odoo) [inline addons/web/static/tests/core/commands/command_service.test.js:169]: to activate the hotkey, one must press 'a' or 'shift-a'? I though 'shift-a', but now I'm not sure. I hope the press helper doesn't add 'shift' automatically.

**Review by aab-odoo — APPROVED** [review]

File: addons/web/static/tests/core/commands/command_service.test.js (hunk 2 above)
Comment (kebeclibre) [reply addons/web/static/tests/core/commands/command_service.test.js:33]: the original file also cleaned that registry. That suite is a generic one and only uses generic providers: not the mail's, not the menus'. The expect.toHaveCount are impacted if other providers are present

File: addons/web/static/tests/core/commands/command_service.test.js (hunk 4 above)
Comment (kebeclibre) [reply addons/web/static/tests/core/commands/command_service.test.js:169]: in this test, there is the pure command A with hotkey "a", and B and C that are hotkeys in the DOM, automatically registered as "alt+[bc]".
press does work as expected.

The original test used triggerHotkey, which, when the second parameter was true, added the alt key stroke artificially

**Comment by kebeclibre** [comment]
@aab-odoo changes made

**Review by aab-odoo — APPROVED** [review]
Thanks for the explanations @kebeclibre 

robodoo r+

**Comment by aab-odoo** [comment]
robodoo rebase-ff

**Pull request closed without merging**